src/cocotb/types/logic_array.py

```diff
@@ -333,6 +333,15 @@ def from_bytes(
             int.from_bytes(value, byteorder=byteorder, signed=False), range=range
         )
 
+    @classmethod
+    def _from_handle(cls, value: str) -> "LogicArray":
+        self = super().__new__(cls)
+        self._value_as_array = None
+        self._value_as_int = None
+        self._value_as_str = value.upper()
```

Comment:
Maybe add a TODO here to guarantee upper-case values already in GPI? Seems like this could give us a bit of performance.

Reply:
Apparently the BinaryValue version of this method didn't uppercase the string, but I'm not sure if that's because it worked with mixed case or if it assumed that the values were already uppercase.